src/main/java/frc/robot/command_groups/Shoot.java

```diff
@@ -0,0 +1,24 @@
+package frc.robot.command_groups;
+
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import frc.robot.subsystems.shooter.Shooter;
+import frc.robot.subsystems.shooter.commands.PrepareShooter;
+
+import java.util.function.DoubleSupplier;
+
+public class Shoot extends ParallelCommandGroup {
+    private Shooter shooter;
+    private DoubleSupplier distance;
```

Comment:
You don't need to store the shooter and the distance outside of the constructor.